src/atc/config_master/table_configurator.py

```diff
@@ -0,0 +1,205 @@
+                if Path(file_name).suffix in [".json", ".yaml", ".yml"]:
+                    with importlib.resources.path(
+                        resource_path, file_name
+                    ) as file_path:
+                        with open(file_path) as file:
+                            update = (
+                                json.load(file)
+                                if file_name.endswith(".json")
+                                else yaml.load(file, Loader=yaml.FullLoader)
+                            )
+                            if not isinstance(update, dict):
+                                raise ValueError(f"document in {file_path} is no dict.")
+
+                            for key, value in update.items():
+                                if self.__is_known_shape(value):
+                                    self.table_names[key] = value
+                                else:
+                                    raise UnknownShapeException(
+                                        f"Object {key} in file {file_path}"
+                                        f" has unexpected shape."
+                                    )
+
+        for key in self.table_names.keys():
+            self.__resolve_key(key)
+
+    def __is_TableDetails_shape(self, value):
+        return set(value.keys()).issubset(TableDetails_keys)
+
+    def __is_TableAlias_shape(self, value):
+        return set(value.keys()).issubset(TableAlias_keys)
+
+    def __is_TableRelDbg_shape(self, value):
+        return set(value.keys()).issubset(TableRelDbg_keys)
+
+    def __is_known_shape(self, value):
+        return (
+            self.__is_TableDetails_shape(value)
+            or self.__is_TableAlias_shape(value)
+            or self.__is_TableRelDbg_shape(value)
+        )
+
+    def __resolve_key(self, key: str):
+        # Handle the case of differentiated release and debug tables
+        if self.__is_TableRelDbg_shape(self.table_names[key]):
+            value: TableRelDbg = self.table_names[key]
+            if self._is_debug:
+                self.table_names[key] = value["debug"]
+            else:
+                self.table_names[key] = value["release"]
+
+        # carry out string substitutions in name and path
+        if self.__is_TableDetails_shape(self.table_names[key]):
+            value: TableDetails = self.table_names[key]
+            if "name" in value:
+                self.table_names[key]["name"] = value["name"].format(
+                    **self.table_arguments
+                )
+            if "path" in value:
+                self.table_names[key]["path"] = value["path"].format(
+                    **self.table_arguments
+                )
+
+    def __get_name_entry(self, table_id: str):
+        entry = self.table_names[table_id]
+        while "alias" in entry:  # allow alias of alias
+            entry = self.table_names[entry["alias"]]
+        return entry
```

Comment:
Could this end up in a infinite loop is two entries has an alias to eachtother?

TableNameA:
     alias: TableNameB


TableNameB:
    alias: TableNameA

I hope that no one does this, but should we address this?